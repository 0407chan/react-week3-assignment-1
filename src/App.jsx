import { useState } from 'react';

import Page from './Page';
import tasksDefault from './tasks';

// 협력 구현 - 종류중의 하나 짝 프로그래밍
// 3분씩 번갈아가면서 프로그래밍

export default function App() {
  const [state, setState] = useState({
    newId: 100,
    taskTitle: '',
    tasks: tasksDefault,
  });

  const { newId, taskTitle, tasks } = state;

  function handleChangeTitle(event) {
    setState({
      ...state,
      taskTitle: event.target.value,
    });
  }

  function handleClickAddTask() {
    setState({
      ...state,
      newId: newId + 1,
      taskTitle: '',
      tasks: [...tasks, { id: newId, title: taskTitle }],
    });
  }

  function handleClickDeleteTask(id) {
    setState({
      ...state,
      tasks: tasks.filter((task) => task.id !== id),
    });
  }

  return (
    <Page
      taskTitle={taskTitle}
      onChangeTitle={handleChangeTitle}
      onClickAddTask={handleClickAddTask}
      tasks={tasks}
      onClickDeleteTask={handleClickDeleteTask}
    />
  );
}
