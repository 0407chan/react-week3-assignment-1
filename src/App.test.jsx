import { fireEvent, screen, render } from '@testing-library/react';

import App from './App';

describe('<App />', () => {
  it('changes taskTitle', () => {
    render(<App />);

    fireEvent.change(screen.getByLabelText('할 일'), { target: { value: '뭐라도 하기' } });

    expect(screen.getByLabelText('할 일')).toHaveAttribute('value', '뭐라도 하기');
  });

  it('creates new task', () => {
    render(<App />);

    fireEvent.change(screen.getByLabelText('할 일'), { target: { value: '뭐라도 하기' } });
    fireEvent.click(screen.getByRole('button', { name: '추가' }));

    screen.getByText('뭐라도 하기');
  });

  it('deletes a task', () => {
    render(<App />);

    fireEvent.change(screen.getByLabelText('할 일'), { target: { value: '뭐라도 하기' } });
    fireEvent.click(screen.getByRole('button', { name: '추가' }));
    fireEvent.click(screen.getByRole('button', { name: '완료' }));

    screen.getByText('할 일이 없어요!');
  });
});
