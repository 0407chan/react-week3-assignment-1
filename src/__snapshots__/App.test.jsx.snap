// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<App/> App 1`] = `
<div>
  <div>
    <h1>
      To-do
    </h1>
    <p>
      <label
        for="input-task-title"
      >
        할 일
      </label>
      <input
        id="input-task-title"
        placeholder="할 일을 입력해 주세요"
        type="text"
        value=""
      />
      <button
        type="button"
      >
        추가
      </button>
    </p>
    <p>
      할 일이 없어요!
    </p>
  </div>
</div>
`;
